File: calculator.cpp
#include <iostream>
using namespace std;

int main() {
    char choice;
    char answer;

    do {
        cout << "=====================================\n";
        cout << "Press 'a' to calculate for speed\n";
        cout << "Press 'b' to calculate for duration\n";
        cout << "Press 'c' to calculate for distance\n";
        cout << "=====================================\n";
        cin >> choice;

        if (choice == 'a') {
            double distance, time, speed;
            cout << "Enter distance in kilometers: ";
            cin >> distance;
            cout << "Enter duration in hours: ";
            cin >> time;
            speed = distance / time;
            cout << "The speed is " << speed << " KPH\n";
        } 
        else if (choice == 'b') {
            double distance, speed, time;
            cout << "Enter distance in kilometers: ";
            cin >> distance;
            cout << "Enter speed in kph: ";
            cin >> speed;
            time = distance / speed;
            cout << "The duration is " << time << " hour/s\n";
        } 
        else if (choice == 'c') {
            double speed, time, distance;
            cout << "Enter speed in kph: ";
            cin >> speed;
            cout << "Enter duration in hours: ";
            cin >> time;
            distance = speed * time;
            cout << "The distance is " << distance << " km\n";
        } 
        else {
            cout << "-------------------------------------\n";
            cout << "ERROR 404: The requested option was not found.\n";
            cout << "Please check your input and try again.\n";
            cout << "-------------------------------------\n";
        }

        cout << "-------------------------------------\n";
        cout << "Do you want to solve again? (y/n): ";
        cin >> answer;

    } while (answer == 'y' || answer == 'Y');

    cout << "Thanks for using the calculator\n";
    return 0;
}

File: Calculator.java
import java.util.Scanner;

public class Calculator {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        char choice;
        char answer;

        do {
            System.out.println("=====================================");
            System.out.println("Press 'a' to calculate for speed");
            System.out.println("Press 'b' to calculate for duration");
            System.out.println("Press 'c' to calculate for distance");
            System.out.println("=====================================");
            choice = sc.next().charAt(0);

            if (choice == 'a') {
                double distance, time, speed;
                System.out.print("Enter distance in kilometers: ");
                distance = sc.nextDouble();
                System.out.print("Enter duration in hours: ");
                time = sc.nextDouble();
                speed = distance / time;
                System.out.println("The speed is " + speed + " KPH");
            } 
            else if (choice == 'b') {
                double distance, speed, time;
                System.out.print("Enter distance in kilometers: ");
                distance = sc.nextDouble();
                System.out.print("Enter speed in kph: ");
                speed = sc.nextDouble();
                time = distance / speed;
                System.out.println("The duration is " + time + " hour/s");
            } 
            else if (choice == 'c') {
                double speed, time, distance;
                System.out.print("Enter speed in kph: ");
                speed = sc.nextDouble();
                System.out.print("Enter duration in hours: ");
                time = sc.nextDouble();
                distance = speed * time;
                System.out.println("The distance is " + distance + " km");
            } 
            else {
                System.out.println("-------------------------------------");
                System.out.println("ERROR 404: The requested option was not found.");
                System.out.println("Please check your input and try again.");
                System.out.println("-------------------------------------");
            }

            System.out.println("-------------------------------------");
            System.out.print("Do you want to solve again? (y/n): ");
            answer = sc.next().charAt(0);

        } while (answer == 'y' || answer == 'Y');

        System.out.println("Thanks for using the calculator");
        sc.close();
    }
}
